openodia/_understandData.py

```diff
@@ -7,20 +8,40 @@
 
 class UnderstandData:
     """Tokenizer implementation"""
-
+    
     @classmethod
-    def word_tokenizer(cls, text):
+    def word_tokenizer(cls,text):
         """Split the text into words"""
-        # TODO: Do not remove the punctuation characters
+        punct = regex.compile(r'(\p{Punctuation})')    
+        x=punct.split(test_data)
+        str1=""
+        for i in x:
+            str1+=i+' '
```

Comment:
This `for` loop increases additional time complexity to the codebase. 
Is there any better way to do it?